elrond-wasm/src/contract_base/wrappers/storage_raw_wrapper.rs

```diff
@@ -0,0 +1,74 @@
+use core::marker::PhantomData;
+
+use elrond_codec::{TopDecode, TopEncode};
+
+use crate::{
+    api::{
+        const_handles::MBUF_TEMPORARY_1, use_raw_handle, ErrorApi, ManagedTypeApi, StorageReadApi,
+        StorageReadApiImpl, StorageWriteApi,
+    },
+    storage::StorageKey,
+    storage_get,
+    storage_get::StorageGetErrorHandler,
+    storage_set,
+    types::{ManagedAddress, ManagedBuffer, ManagedType},
+};
+
+#[derive(Default)]
+pub struct StorageRawWrapper<A>
+where
+    A: StorageReadApi + ManagedTypeApi + ErrorApi,
+{
+    _phantom: PhantomData<A>,
+}
+
+impl<A> StorageRawWrapper<A>
+where
+    A: StorageReadApi + StorageWriteApi + ManagedTypeApi + ErrorApi,
+{
+    pub fn new() -> Self {
+        StorageRawWrapper {
+            _phantom: PhantomData,
+        }
+    }
+
+    /// Reads storage from the given key, and deserializes the value to the provided type.
+    ///
+    /// Use only if really necessary, storage mappers should be preferred.
+    #[inline]
+    pub fn read<T: TopDecode>(&self, storage_key: &StorageKey<A>) -> T {
```

Comment:
Simple to implement and it helps with syntax: having the key of generic type K, where StorageKey<A>: From<K>, and calling .into() inside.

Also implement `StorageKey` `From` `&str`.

We can then write `self.storage_raw().read("my_key")` and `self.storage_raw().read(managed_buffer)`